tree/ntuple/v7/inc/ROOT/RNTupleView.hxx

```diff
@@ -203,6 +227,26 @@ public:
    {
       return fField.MapV(clusterIndex, nItems);
    }
+
+   void Bind(std::shared_ptr<T> objPtr)
+   {
+      static_assert(
+         UserProvidedAddress,
+         "Only views which were created with an external memory location at construction time can be bound to a "
+         "different memory location afterwards. Call the RNTupleReader::GetView overload with a shared_ptr.");
+      fValue.Bind(objPtr);
+   }
+
+   void BindRawPtr(T *rawPtr)
+   {
+      static_assert(
+         UserProvidedAddress,
+         "Only views which were created with an external memory location at construction time can be bound to a "
+         "different memory location afterwards. Call the RNTupleReader::GetView overload with a shared_ptr.");
+      fValue.BindRawPtr(rawPtr);
+   }
+
+   void EmplaceNew() { fValue.EmplaceNew(); }
```

Comment:
I was wondering about this and I didn't add it because I think in both scenarios (if the user provides an external memory address or not), `EmplaceNew` would still be valid, right? The user is explicitly giving back the `RNTupleView` control over the target memory to read values into. In the case of a mappable value without user-provided address, this would involve creating an RValue which is then not used, fair enough, that should not be harmful? In the case of a mappable value with user provided address, that new RValue will be used internally to read data. What do you think ?

Reply:
That sounds fair. I think we should still throw a static assert for zero-copy views (mappable type, no user address).